src/app/dashboard-widgets/create-work-item-widget/create-work-item-widget.component.ts

```diff
@@ -14,41 +14,73 @@ import { filterOutClosedItems, WorkItemsData } from '../../shared/workitem-utils
 export class CreateWorkItemWidgetComponent implements OnInit {
 
   @Input() userOwnsSpace: boolean;
+  @Input() loggedInUser: User;
+  @Input() currentSpace: Space;
 
-  private _myWorkItems: ConnectableObservable<WorkItem[]>;
-  myWorkItemsCount: Observable<number>;
+  private _myWorkItems: WorkItem[];
+  myWorkItemsCount: number;
   contextPath: Observable<string>;
+  subscriptions: Subscription[] = [];
 
   constructor(
     private filterService: FilterService,
     private workItemService: WorkItemService,
-    private userService: UserService,
     private contexts: Contexts
   ) {}
 
   ngOnInit() {
     this.contextPath = this.contexts.current.pipe(map(context => context.path));
-    this._myWorkItems = this.userService
-      .getUser().pipe(
-      tap(() => this.workItemService.buildUserIdMap()),
-      switchMap(() => this.userService.loggedInUser),
-      map(user => this.filterService.queryBuilder('assignee', this.filterService.equal_notation, user.id)),
-      switchMap(filters => this.workItemService
-        .getWorkItems(100000, {expression: filters})),
-      map((val: WorkItemsData) => val.workItems),
-      map(workItems => filterOutClosedItems(workItems)),
-      // Resolve the work item type
-      tap(workItems => workItems.forEach(workItem => this.workItemService.resolveType(workItem))),
-      tap(workItems => workItems.forEach(workItem => this.workItemService.resolveAreaForWorkItem(workItem))),
-      // MUST DO creator after area due to bug in planner
-      tap(workItems => workItems.forEach(workItem => this.workItemService.resolveCreator(workItem))),
-      publishReplay(1)) as ConnectableObservable<WorkItem[]>;
-    this._myWorkItems.connect();
-    this.myWorkItemsCount = this._myWorkItems.pipe(map(workItems => workItems.length));
+    this.getCurrentSpaceWorkItems();
   }
 
-  get myWorkItems(): Observable<WorkItem[]> {
+  ngOnChanges() {
+    this.getCurrentSpaceWorkItems();
+  }
+
+  ngOnDestroy(): void {
+    this.subscriptions.forEach(sub => {
+      sub.unsubscribe();
+    });
+  }
+
+  get myWorkItems(): WorkItem[] {
     return this._myWorkItems;
   }
 
+  getCurrentSpaceWorkItems() {
+    const assigneeQuery = this.filterService.queryJoiner(
+      {},
+      this.filterService.and_notation,
+      this.filterService.queryBuilder(
+        'assignee', this.filterService.equal_notation, this.loggedInUser.id
+      )
+    );
+    const spaceQuery = this.filterService.queryBuilder(
+      'space', this.filterService.equal_notation, this.currentSpace.id
+    );
+    const filters = this.filterService.queryJoiner(
+      assigneeQuery, this.filterService.and_notation, spaceQuery
+    );
+    this.subscriptions.push(
+      this.workItemService
+       .getWorkItems(100000, {expression: filters}).pipe(
+          map((val: WorkItemsData) => val.workItems),
+          map(workItems => filterOutClosedItems(workItems)),
```

Comment:
It'd be appreciated it the type definitions could be added here (and elsewhere in this function chain too), e.g.,
`map(workItems: WorkItem[]) => [...]`